main.cpp

```diff
@@ -43,15 +45,73 @@ struct Point
     float x{0}, y{0};
 };
 
+//Wrapper/////////////////////////////////////////
+
 template<typename Type>
 struct Wrapper
 {
     Wrapper(Type&& t) : val(std::move(t)) 
     { 
         std::cout << "Wrapper(" << typeid(val).name() << ")" << std::endl; 
     }
+
+    void print()
+    {
+        if constexpr(std::is_same<decltype(val), double>::value)
+        {
+            if(val == 3.5)
+            {
+                std::cout << "Value: "; 
+                std::cout << "Wrapper::print(" << val << ")" << std::endl; 
+                std::cout << "------------------------" << std::endl;
+            }
+        } 
```

Comment:
This `print()` function should be generic for any `Type` you pass to this wrapper.
- [ ] remove the type and value filtering.